server/rest/src/main/java/org/infinispan/rest/resources/CacheResourceV2.java

```diff
@@ -195,6 +200,27 @@ public Invocations getInvocations() {
       }, invocationHelper.getExecutor());
    }
 
+   private CompletionStage<RestResponse> entries(RestRequest request) {
+      String cacheName = request.variables().get("cacheName");
+      String limitParam = request.getParameter("limit");
+      int limit = limitParam == null ? 100 : Integer.parseInt(limitParam);
+
+      Cache<?, ?> cache = invocationHelper.getRestCacheManager().getCache(cacheName, APPLICATION_JSON, APPLICATION_JSON, request);
+      if (cache == null)
+         return notFoundResponseFuture();
+
+      // Streaming over the cache is blocking
+      return CompletableFuture.supplyAsync(() -> {
+         NettyRestResponse.Builder responseBuilder = new NettyRestResponse.Builder();
+         responseBuilder.entity(cache.entrySet().stream().limit(limit).map(this::mapToEntryInfo)
+               .sorted(Comparator.comparingLong(o -> o.created)).map(EntryInfo::toJson).collect(Collectors.toList()));
```

Comment:
There's are some issues here: if the supplied limit param is too high, it will cause memory explosion in the server since all the entries will be bought to a single node. If the limit is ```100``` or very low, it will only be able to return those 100 items and not the others. So the JIRA was about iteration, and we're not iterating here. 

If we want to return all entries, OTOH, it should be done the same as the ```keys``` operation where it is streamed taking advantage of ```CHUNKED``` Http encoding.

Reply:
One idea of how an iteration API could work, is to keep the cache's iteration alive in the server, return a unique identifier to the client, and the client would calls ```next``` multiple times using that identifier until the entries are exhausted. Something along the lines:

Start the iteration:
```POST /rest/v2/caches/my-cache/entries``` returns an unique id

Iterate (calls N times)
```GET /rest/v2/caches/my-cache/entries?scroll-id=<id>```


As this would imply only a single node in the server would have an opened iterator for a certain id, some mechanism of either redirecting the client to the right server, or redirecting the request to the appropriate server would be required

Reply:
Things get more complicated in the above strategy when the server where the iterator is opened crashes, so we lose all the state. In Hot Rod, we solved this by keeping track of the keys/segments that were already iterated in the client, detecting the crash and restarting the iteration from another node, skipping those entries already seen. 
In REST, this can't be done, as we can't keep state in the client since we don't control it.

Reply:
I would like to support 2 different methods here. 
Streaming like keys (that said I was unable to 'stream' those keys for now locally)
Getting a small sample of entries at once (useful for small caches)
This is a completely preview PR and WIP, but there is demand for this for 12 for the console, and I believed the REST API supported this but it only supports keys, and the streaming does not seem to work as expected or probably is me that does not get how to make it work

Reply:
> Things get more complicated in the above strategy when the server where the iterator is opened crashes, so we lose all the state. In Hot Rod, we solved this by keeping track of the keys/segments that were already iterated in the client, detecting the crash and restarting the iteration from another node, skipping those entries already seen.
> In REST, this can't be done, as we can't keep state in the client since we don't control it.

Maybe would be interesting opening the opportunity to have some data on the server ? a cache for user/console data?
In REST we can't, but we can keep track of stuff in the navigator.

Reply:
> Streaming like keys (that said I was unable to 'stream' those keys for now locally)

What issue did you have? You can test with cURL or the CLI and it should work.

> REST API supported this but it only supports keys,

REST only supports streamed (HTTP Chunked) ```keys```. The other method was removed since it was extremely inefficient 

> Getting a small sample of entries at once (useful for small caches)

With a cursor/iteration this should be doable: just start an iteration and call get once

Reply:
The ```keys``` operation always return all keys, the ```batch``` parameter is internal, and is used to configure the amount of entries that will be kept in memory at once, it does not affect the response.

Here's my suggestion:

*  Create another JIRA, as ISPN-9738 is about iteration
*  In that new JIRA, we can provide an ```entries``` or ```values``` endpoint that only does streaming, just like ```keys```. We should avoid loading all entries in memory based on user supplied params, or we're back to square one
*  An extra param ```?limit=x``` would be added to those operations to limit the amount returned, but document that pagination will not be supported.

Reply:
This works for me @gustavonalle

Reply:
ISPN-12444